### PR TITLE
fix acpi tables being overwritten 

### LINKED ISSUE #2546: ACPI Tables overriden by Memory Manager
Cause bochs errors, for some reasons works on VMWare

### PR DESCRIPTION
**Pull request by AsertCreator**
it should work but i didnt test yet and dont now how. well i fixed acpi table overlapping by making memory allocator to not allocate "unusable" addresses. push to feature/apic. 

fixes #2546 

- [x] works on VMware - well at least memory manager works well, apic may work but i didn't check
- [x] works on qemu - mm is working, but os utilities are not working correctly. maybe it is my fault cos i wrote pretty bad code
- [ ] works on bochs - awaits testing. it doesnt work damnit

### CODE REVIEW

**Comment by AsertCreator** [comment]
hmm, why appveyor test doesnt start?

**Comment by terminal-cs** [comment]
> hmm, why appveyor test doesnt start?

Because you aren't pushing to the main branch

**Comment by AsertCreator** [comment]
oh okay

**Comment by valentinbreiz** [comment]
Thanks for the fix it could be merged if working but it's a tricky way to work around the bug... the actual reason why ACPI tables can be overwritten is because the multiboot2 memory map is kind of broken

**Comment by AsertCreator** [comment]
oh when i run it in appveyor kernel crashes after ps/2 controller initialization

**Comment by TheFocusMan** [comment]
> oh when i run it in appveyor kernel crashes after ps/2 controller initialization

yep it is happen to me too in Fat File Test it is can be beacuse the py.yml, before the file added appveyor want stop on timeout

**Comment by AsertCreator** [comment]
> > oh when i run it in appveyor kernel crashes after ps/2 controller initialization
> 
> yep it is happen to me too in Fat File Test it is can be beacuse the py.yml, before the file added appveyor want stop on timeout

apparently it says that kernel crashed at nullreferenceexception. maybe something wrong with bochs? cause it works on vmware and qemu. or maybe it didnt get enough memory and RAT just overwritten acpi tables or something? it means we need to give bochs more memory.

**Comment by quajak** [comment]
> > > oh when i run it in appveyor kernel crashes after ps/2 controller initialization
> > 
> > 
> > yep it is happen to me too in Fat File Test it is can be beacuse the py.yml, before the file added appveyor want stop on timeout
> 
> apparently it says that kernel crashed at nullreferenceexception. maybe something wrong with bochs? cause it works on vmware and qemu. or maybe it didnt get enough memory and RAT just overwritten acpi tables or something? it means we need to give bochs more memory.

Can you check the map file for in which method it crashes with the null reference exception? We shouldnt be running into memory issues

**Comment by AsertCreator** [comment]
```024a3771 l       .text	00000000 SystemInt32SystemCollectionsGenericEqualityComparer1CosmosHALPCIDeviceIndexOfCosmosHALPCIDevicearrayCosmosHALPCIDeviceSystemInt32SystemInt32.IL_0012.AfterAddressCheck```
its should be it. i cant understand how il2cpu encodes method names but i can tell that something is for `EqualityComparer`

**Review by MishaTY — CHANGES_REQUESTED** [review]

File: Build/limine/boot/limine.cfg [hunk 1]
```diff
@@ -0,0 +1,12 @@
+DEFAULT_ENTRY=1
```
[the remark this answers is not in the document]
Comment (AsertCreator) [reply Build/limine/boot/limine.cfg:1]: i did this with rebase. virtually, there are changes. practically there are no new changes

**Comment by AsertCreator** [comment]
okay, this wont work. if someone has time, please fix this

**Pull request closed without merging**